Caliptra core 2.0 HW spec updates

bharatpillilli opened the pull request.

calebofearth commented on docs/CaliptraHardwareSpecification.md:
```diff
@@ -18,6 +19,18 @@ This document provides definitions and requirements for a Caliptra cryptographic
+    * UDS programming
+    * Read logic for Secret Fuses
+    * Streaming Boot Support
+    * DMA Support
```
Rewording suggestions:
```
* AXI subordinate replaces APB interface of Caliptra 1.x hardware
* SHA Accelerator functionality now available exclusively over mailbox
    * SHA Accelerator adds new SHA save/restore functionality
* Adams Bridge Dilithium/ML-DSA (refer to [Adams bridge spec](https://github.com/chipsalliance/adams-bridge/blob/main/docs/Adams\ bridge_HardwareSpecs.docx))
* Subystem mode support (refer to [Subsystem Specification](https://github.com/chipsalliance/caliptra-ss/blob/main/docs/Caliptra%202.0%20Subsystem%20Specification%201.pdf) for details)
    * AXI Manager with DMA support
    * Manufacturing and Debug Unlock
    * UDS programming
    * Read logic for Secret Fuses
    * Streaming Boot Support
* RISC-V core PMP support
```

calebofearth commented on docs/CaliptraHardwareSpecification.md:
```diff
@@ -457,9 +469,9 @@ When the RV core wakes up, all clocks are enabled. However, when the core is hal
-Activity on the APB interface only wakes up the SoC IFC clock. All other clocks remain off until any other condition is met or the core exits the halt state.
+Activity on the AXI interface only wakes up the SoC IFC clock. All other clocks remain off until any other condition is met or the core exits the halt state.
 
 | Cpu_halt_status | PSEL | Generic input wires <br>\|\| fatal error <br>\|\| debug/scan mode  <br> \|\|JTAG access | Expected behavior |
```
Access table no longer uses "PSEL" - it's a combination of arvalid and awvalid and "s_axi_active", which tracks whether requests are outstanding or not.

bharatpillilli replied:
what should we call it?

calebofearth commented on docs/CaliptraHardwareSpecification.md:
```diff
@@ -20,17 +20,18 @@ This document provides definitions and requirements for a Caliptra cryptographic
+* SHA Accelerator functionality now available exclusively over mailbox
+    * SHA Accelerator adds new SHA save/restore functionality
+* Adams Bridge Dilithium/ML-DSA (refer to [Adams bridge spec](https://github.com/chipsalliance/adams-bridge/blob/main/docs/Adams\ bridge_HardwareSpecs.docx))
+* Subystem mode support (refer to [Subsystem Specification](https://github.com/chipsalliance/caliptra-ss/blob/main/docs/Caliptra%202.0%20Subsystem%20Specification%201.pdf) for details)
```
Typo here - my bad. "Subystem" -> "Subsystem"

calebofearth commented on docs/CaliptraHardwareSpecification.md:
```diff
@@ -18,6 +19,19 @@ This document provides definitions and requirements for a Caliptra cryptographic
+* AXI subordinate replaces APB interface of Caliptra 1.x hardware
+* SHA Accelerator functionality now available exclusively over mailbox
+    * SHA Accelerator adds new SHA save/restore functionality
+* Adams Bridge Dilithium/ML-DSA (refer to [Adams bridge spec](https://github.com/chipsalliance/adams-bridge/blob/main/docs/Adams\ bridge_HardwareSpecs.docx))
```
This one is also a typo from an older suggestion.... instead of "Adams\ bridge_HardwareSpecs..." it should be "Adams%20bridge_HardwareSpecs".
Broken link.

calebofearth commented on docs/CaliptraHardwareSpecification.md:
```diff
@@ -474,13 +474,13 @@ When the RV core wakes up, all clocks are enabled. However, when the core is hal
 Activity on the AXI interface only wakes up the SoC IFC clock. All other clocks remain off until any other condition is met or the core exits the halt state.
 
-| Cpu_halt_status | PSEL | Generic input wires <br>\|\| fatal error <br>\|\| debug/scan mode  <br> \|\|JTAG access | Expected behavior |
+| Cpu_halt_status | axi_active | Generic input wires <br>\|\| fatal error <br>\|\| debug/scan mode  <br> \|\|JTAG access | Expected behavior |
```
let's do s_axi_active to distinguish from the manager.

calebofearth commented on docs/CaliptraHardwareSpecification.md:
```diff
@@ -474,13 +474,13 @@ When the RV core wakes up, all clocks are enabled. However, when the core is hal
 
 Activity on the AXI interface only wakes up the SoC IFC clock. All other clocks remain off until any other condition is met or the core exits the halt state.
```
Activity on the AXI subordinate interface...

calebofearth commented on docs/CaliptraHardwareSpecification.md:
```diff
@@ -92,15 +107,14 @@ The following table shows the memory map address ranges for each of the IP block
 | SHA256                              | 13        | 32 KiB       | 0x1002_8000   | 0x1002_FFFF |
+| ML-DSA                              | 17        | 64 KiB       | 0x1003_0000   | 0x1003_FFFF |
 
 #### Peripherals subsystem
```
Let's also update the interrupt vector list here: 
https://github.com/chipsalliance/caliptra-rtl/blob/6cd75c8b6d71e87ef9a0d1ba8bc0596651183770/docs/CaliptraHardwareSpecification.md#external-interrupts

The latest and greatest list of interrupt vectors is in [this file](https://github.com/chipsalliance/caliptra-rtl/pull/600/files#diff-052012e084cb1592561733e71302a9cbfebf3d3cfe4a422856af1049f13fad33)

calebofearth reviewed (changes requested):
Fix this typo? [dilitium](https://github.com/chipsalliance/caliptra-rtl/blob/bpillilli-HWspec-2p0-0p5-update/docs/CaliptraHardwareSpecification.md#adams-bridge---dilitium-ml-dsa)

Remove the entire section on [UART/QSPI](https://github.com/chipsalliance/caliptra-rtl/blob/bpillilli-HWspec-2p0-0p5-update/docs/CaliptraHardwareSpecification.md#peripherals-subsystem-1)?

Is the [KeyVault interface update](https://github.com/chipsalliance/caliptra-rtl/blob/bpillilli-HWspec-2p0-0p5-update/docs/CaliptraHardwareSpecification.md#key-vault-cryptographic-functional-block) known for ML-DSA?

Should [these integration spec sections](https://github.com/chipsalliance/caliptra-rtl/blob/main/docs/CaliptraIntegrationSpecification.md#integration-parameters) also be updated (remove qspi/uart)?

bharatpillilli commented:
> Fix this typo? [dilitium](https://github.com/chipsalliance/caliptra-rtl/blob/bpillilli-HWspec-2p0-0p5-update/docs/CaliptraHardwareSpecification.md#adams-bridge---dilitium-ml-dsa)
> 
> Remove the entire section on [UART/QSPI](https://github.com/chipsalliance/caliptra-rtl/blob/bpillilli-HWspec-2p0-0p5-update/docs/CaliptraHardwareSpecification.md#peripherals-subsystem-1)?
> 
> Is the [KeyVault interface update](https://github.com/chipsalliance/caliptra-rtl/blob/bpillilli-HWspec-2p0-0p5-update/docs/CaliptraHardwareSpecification.md#key-vault-cryptographic-functional-block) known for ML-DSA?
> 
> Should [these integration spec sections](https://github.com/chipsalliance/caliptra-rtl/blob/main/docs/CaliptraIntegrationSpecification.md#integration-parameters) also be updated (remove qspi/uart)?


- Fixed the typo
- Removed QSPI/UART from main spec
- once KV work is done, @Nitsirks make this update
- We can do integ spec update when the QSPI/UART are truly removed (soon by you :-))

bharatpillilli commented:
@Nitsirks - let me know if you want me to update the KV thing @calebofearth mentioned above and I can do that since you may already know what you are doing and launch this PR. If you want to do it as a part of your PR, I will send this in so that you can make doc change on top of it without having conflicts

calebofearth commented on docs/CaliptraHardwareSpecification.md:
```diff
@@ -18,6 +19,19 @@ This document provides definitions and requirements for a Caliptra cryptographic
+## Key Caliptra Core 2.0 Changes
+* AXI subordinate replaces APB interface of Caliptra 1.x hardware
+* SHA Accelerator functionality now available exclusively over mailbox
+    * SHA Accelerator adds new SHA save/restore functionality
```
Should we wait to list save/restore function for SHA Accelerator? It is not yet enabled.

calebofearth replied:
Maybe just a separate bullet listing it for the sha512 block instead

bharatpillilli replied:
just a spec; PR will come through as a part of Freeze, Mojtaba is there

calebofearth commented on docs/CaliptraHardwareSpecification.md:
```diff
@@ -161,14 +174,16 @@ Vector 0 is reserved by the RISC-V processor and may not be used, so vector assi
 | SHA512 (Notifications)                              | 10               | 7                                               |
 | SHA256 (Errors)                                     | 11               | 8                                               |
 | SHA256 (Notifications)                              | 12               | 7                                               |
-| QSPI (Errors)                                       | 13               | 4                                               |
```
Change to RESERVED instead of removing?

bharatpillilli replied:
does it matter?

bharatpillilli commented on docs/CaliptraHardwareSpecification.md:
```diff
@@ -18,6 +19,19 @@ This document provides definitions and requirements for a Caliptra cryptographic
 
 For information on the Caliptra Core, see the [High level architecture](https://chipsalliance.github.io/Caliptra/doc/Caliptra.html#high-level-architecture) section of [Caliptra: A Datacenter System on a Chip (SoC) Root of Trust (RoT)](https://chipsalliance.github.io/Caliptra/doc/Caliptra.html).
 
+## Key Caliptra Core 2.0 Changes
```
I didnt add AES engine explicitly since we are still settling on it but will add the other two
